### PR TITLE
Explicitly set next expected call in case task was never executed or …

### PR DESCRIPTION
**Pull request by itshane**
…scheduled.

Fixes ZPS-960

### CODE REVIEW

File: Products/ZenCollector/scheduler.py [hunk 1]
```diff
@@ -505,12 +508,23 @@ def getTasksForConfig(self, configId):
...
         loopingCall = self._loopingCalls.get(taskName, None)
         if loopingCall:
             return loopingCall._expectNextCallAt
 
+    def setNextExpectedRun(self, taskName, taskInterval):
+        """
+        Set the next expected execution time for given task.
+        """
+        loopingCall = self._loopingCalls[taskName]
+        loopingCall._expectNextCallAt = time.time() + taskInterval
```
Comment (achow-zenoss) [inline Products/ZenCollector/scheduler.py]: Could loopingCall nil? as check in line #514?

Comment (itshane) [inline Products/ZenCollector/scheduler.py]: It shouldn't be the case according to the execution order, however, I've updated the branch, please recheck.

**Comment by itshane** [comment]
retest this

**Comment by itshane** [comment]
retest this

**Pull request merged**